[#14] AWS S3 환경설정

choitaehoon opened the pull request:
- AWS의 key값 분리 후 gitignore 적용
- AmazonS3 빈으로 등록

choitaehoon committed:
[#14] AWS S3 환경설정

- AWS의 key값 분리 후 gitignore 적용
- AmazonS3 빈으로 등록

choitaehoon committed:
[#14] AWS S3 환경설정

- javax.annotation-api 종속성 추가

f-lab-dev commented on src/main/java/com/social/instagram/config/AwsS3Config.java:
```diff
@@ -0,0 +1,40 @@
+    외부 환경 변수의 경로를 지정하고 외부 환경 변수의 값을 @Value에 주입하여 사용 할 수 있다.
+*/
+@Configuration
+@PropertySource("classpath:application-aws.properties")
```
classpath라는 것은 결국 프로젝트 내부에 properties 파일이 있다는건데 그것보다는 spring에서 property를 주입하는 방식이 무엇무엇이 있는지 알아보시고, 그것을 고려해서 따로 설정해주셔도 될 것 같아요~

choitaehoon replied:
Springboot에서는 `ApplicationContext`가 생성되기 전 환경 변수를 사용하고 있는 **무엇** 인가를 통해 `Auto configuration`에 영향을 미치고 있어 권장하는 방법이 아니군요. 무엇이 있는지 알아보고 고려해서 설정하겠습니다

choitaehoon committed:
[#14] AWS S3 환경설정

- PropertySource 제거 후 EnvironmentPostProcessor 생성
- aws 환경설정 예외 처리
- aws 환경 설정, 메시지에 관련된 상수 생성

choitaehoon committed:
[#14] AWS S3 환경설정

- gitignore application-aws.yml로 변경

choitaehoon committed:
[#14] AWS S3 환경설정

- 주석제거

f-lab-dev commented on src/main/java/com/social/instagram/config/AwsS3EnvironmentPostProcessor.java:
```diff
@@ -0,0 +1,51 @@
+import static com.social.instagram.util.aws.AwsResponseConstants.ERROR_MESSAGE_AWS_YML_FAIL_LOAD;
+import static com.social.instagram.util.aws.AwsResponseConstants.ERROR_MESSAGE_AWS_YML_NOT_PATH;
+
+public class AwsS3EnvironmentPostProcessor implements EnvironmentPostProcessor {
```
굳이 이렇게 커스터마이징 안해주셔도 스프링에서 property를 외부에서 주입할 수 있도록 지원하는 것들은 매우 많습니다. 굳이 지원하는게 있는데도 이렇게 커스터마이징하는 것은 오버엔지니어링 같네요~

`inject spring property` 등의 키워드로 검색해보시면 연관검색어에도 관련 내용이 많이 나올 것 같습니다.

choitaehoon committed:
[#14] AWS S3 환경설정

- EnvironmentPostProcessor 제거
- application.yml 파일안에 aws include

f-lab-dev commented:
이런 것도 보시면 좋겠네요~
https://www.baeldung.com/spring-properties-file-outside-jar

choitaehoon commented:
이 PR merge 해도 되나여~?

f-lab-dev reviewed (approved):
네 다만 property들을 외부에서 주입할 수 있는 방법들에 대해서는 숙지해두셨다가 나중에 배포할때 쓰셔야합니다~
